Sources/Client/Model/Channel/Channel.swift

```diff
@@ -297,23 +303,37 @@ extension Channel {
     /// A channel name.
     public var name: String? {
         get {
-            extraData?.name
+            print("###", "Getting name for channel \(id)")
+            if nameAndImageForCurrentUser.name == nil, let namingStrategy = namingStrategy {
+                print("###", "Generating name for channel \(id)")
+                // Save generated name to nameAndImageForCurrentUser since there isn't any
+                nameAndImageForCurrentUser.name = namingStrategy.name(for: User.current, members: members.map(\.user))
```

Comment:
I don't think we need/should store the name. It's OK to compute it every time needed.